remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala

```diff
@@ -511,28 +500,34 @@ class NettyTransport(val settings: NettyTransportSettings, val system: ExtendedA
               s"Unknown local address type [${newServerChannel.localAddress().getClass.getName}]")
         }
       } catch {
-        case NonFatal(e) => {
-          log.error("failed to bind to {}, shutting down Netty transport", address)
+        case NonFatal(e) =>
+          log.error("failed to bind to host:{} port:{}, shutting down Netty transport", settings.BindHostname, bindPort)
           try {
             shutdown()
-          } catch { case NonFatal(_) => } // ignore possible exception during shutdown
-          throw e
-        }
+          } catch {
+            case NonFatal(_) =>
+          } // ignore possible exception during shutdown
+          throw e;
       }
-    }
+    })
   }
 
   // Need to do like this for binary compatibility reasons
   private[pekko] def boundAddress = boundTo
 
+  private def extractHostAndPort(addr: Address): (String, Int) = addr match {
+    case Address(_, _, Some(host), Some(port)) => (host, port)
+    case _                                     => throw new IllegalArgumentException(s"Address [$addr] does not contain host or port information.")
+  }
+
   override def associate(remoteAddress: Address): Future[AssociationHandle] = {
     if (!serverChannel.isActive) Future.failed(new NettyTransportException("Transport is not bound"))
     else {
       val bootstrap: ClientBootstrap = outboundBootstrap(remoteAddress)
 
       (for {
-        socketAddress <- addressToSocketAddress(remoteAddress)
-        readyChannel <- NettyFutureBridge(bootstrap.connect(socketAddress)).map { channel =>
+        (host, port) <- Future.fromTry(Try(extractHostAndPort(remoteAddress)))
```

Comment:
this seems bad to wrap the extractHostAndPort in a Try and a Future. Why do this call before the for comprehension so you won't need to wrap the result.

Reply:
Because it throws